src/buttons/radio.ts

```diff
@@ -133,7 +139,13 @@ export class NgbRadio implements OnDestroy {
 
   ngOnDestroy() { this._group.unregister(this); }
 
-  onChange() { this._group.onRadioChange(this); }
+  onChange() { this._group.onRadioChange(this.value); }
+
+  onClick() {
```

Comment:
So, I must say that I'm super-worried about the sequence of `click` / `change` events here - in this code you are assuming that `click` will fire before `change` but this not the case in all browsers, AFAIK.

Also, the current code would trigger model mutation twice and I don't think this is right.

Generally speaking we should make sure that only one of `click` / `change` is processed.

Reply:
thanks for the feedback.

yeah I had some bad feelings as well, but didn't want to remove the onchange, but it seems like we have to remove the onchange to make it work. 

I didn't know that browser differ so much in that regard.

Reply:
> but it seems like we have to remove the onchange to make it work.

We can't do this as it would kill keyboard support, if I'm not mistaken